src/OpenLoco/Vehicles/VehicleBody.cpp

```diff
@@ -960,40 +960,41 @@ namespace OpenLoco::Vehicles
             var_55 = 0;
         }
 
-        bool itemFound = false;
+        bool stationFound = false;
 
-        // Looking for a bridge? or something on top
+        // Looking for a station
         if (steam_obj->var_08 & (1 << 2))
         {
             auto tile = Map::TileManager::get(frontBogie->tile_x, frontBogie->tile_y);
 
             for (auto& el : tile)
             {
-                if (itemFound && !(el.isGhost() || el.isFlag5()))
+                if (stationFound && !(el.isGhost() || el.isFlag5()))
                 {
                     break;
                 }
                 else
                 {
-                    itemFound = false;
+                    stationFound = false;
                 }
                 auto track = el.asTrack();
                 if (track == nullptr)
                     continue;
                 if (track->baseZ() != frontBogie->tile_base_z)
                     continue;
-                if (track->unkZ() != smokeLoc.z)
+                if (track->trackId() != ((frontBogie->var_2C >> 3) & 0x3F))
```

Comment:
Can we make accessor functions for var_2C as well, so the implementation can easily be changed later?

(If they are already used in several places, this could be best left to be refactored in a separate PR.)

Reply:
It's used in a ton of places. I've a separate pr for that but it's more complex than i thought